system-addon/test/unit/lib/Screenshots.test.js

```diff
@@ -49,6 +49,34 @@ describe("Screenshots", () => {
     });
   });
 
+  describe("#maybeCacheScreenshot", () => {
+    let link;
+    beforeEach(() => {
+      link = {__sharedCache: {updateLink: sinon.stub()}};
+      sandbox.stub(Screenshots, "getScreenshotForURL");
+    });
+    it("should call getScreenshotForURL", () => {
+      Screenshots.maybeCacheScreenshot(link, "mozilla.com", "image", sinon.stub());
+
+      assert.calledOnce(Screenshots.getScreenshotForURL);
+      assert.calledWithExactly(Screenshots.getScreenshotForURL, "mozilla.com");
+    });
+    it("should not call getScreenshotForURL if a fetch is in progress", () => {
+      link.__sharedCache.fetchingScreenshot = sinon.stub();
```

Comment:
You should be able to test this without reaching into the details with two calls to `maybeCacheScreenshot` of the same `link`